tests/RobotFramework/tasks/E2E_Tutorials/Install.robot

```diff
@@ -0,0 +1,102 @@
+   Uninstall ThinEdgeIO
+
+Optional: Linux distributions without systemd curl
+   ${OUTPUT}    Execute Command    curl -fsSL https://thin-edge.io/install-services.sh | sh -s    ignore_exit_code=True
+   #Not verifiing this test step because the test running in container already exists: tests/RobotFramework/tests/installation/install_on_linux.robot
+   #Checking only that the link is correct
+   Should Contain    ${OUTPUT}    Welcome to the thin-edge.io community!
+
+Optional: Linux distributions without systemd wget
+   ${OUTPUT}    Execute Command    wget -O - https://thin-edge.io/install-services.sh | sh -s
+   #Not verifiing this test step because the test running in container already exists: tests/RobotFramework/tests/installation/install_on_linux.robot
+   #Checking only that the link is correct
+   Should Contain    ${OUTPUT}    Welcome to the thin-edge.io community!
+
+Manual repository setup and installation running with sudo
+   ${OUTPUT}    Execute Command    curl -1sLf 'https://dl.cloudsmith.io/public/thinedge/tedge-release/setup.deb.sh' | sudo bash
+   Should Contain    ${OUTPUT}    The repository has been installed successfully - You're ready to rock!
+   ${log}    Execute Command    sudo apt update
+   Check repository creation
+   ${log}    Execute Command    sudo apt-get install -y tedge-full
+   Verify ThinEdgeIO is installed
+   Uninstall ThinEdgeIO
+   Remove created repository
+
+Manual repository setup and installation running as root
+   ${OUTPUT}    Execute Command    sudo su -c "whoami && curl -1sLf 'https://dl.cloudsmith.io/public/thinedge/tedge-release/setup.deb.sh' | bash && apt update && apt-get install -y tedge-full"
+   Should Contain    ${OUTPUT}    root
+   Should Contain    ${OUTPUT}    The repository has been installed successfully - You're ready to rock!
+   Verify ThinEdgeIO is installed
+   Uninstall ThinEdgeIO
+   Remove created repository
+     
+Install via tarball
+   ${log}    Execute Command    curl -fsSL https://thin-edge.io/install.sh | sh -s -- --package-manager tarball
+   Verify ThinEdgeIO is installed
+   Uninstall ThinEdgeIO
+
+
+*** Keywords ***
+
+Verify ThinEdgeIO is installed
+   ${OUTPUT}    Execute Command    tedge --help
+   Should Contain    ${OUTPUT}    tedge is the cli tool for thin-edge.io
+   Log    ThinEdgeIO was successfully installed
+
+Uninstall ThinEdgeIO
+   Transfer To Device    ${CURDIR}/uninstall-thin-edge_io.sh    /home/pi/uninstall-thin-edge_io.sh
+   Execute Command    chmod a+x uninstall-thin-edge_io.sh
+   Execute Command    ./uninstall-thin-edge_io.sh purge
+   Log    Succesfull uninstalled with purge
+
+#Verify ThinEdgeIO is uninstalled
+   ${OUTPUT}    Execute Command    tedge --help    ignore_exit_code=True
+   Should Not Contain    ${OUTPUT}    tedge is the cli tool for thin-edge.io
+
+Check repository creation
+   ${OUTPUT}    Execute Command    ls /etc/apt/sources.list.d/
+   Should Contain    ${OUTPUT}    thinedge-tedge-release.list
```

Comment:
```suggestion
```

I would exclude the check for an explicit `*.list` file as the filename is not part of the public interface. The `apt-cache search tedge` is enough to validate if the thin-edge.io apt source list was added successfully.